pkg/daemon/daemon.go

```diff
@@ -1120,6 +1192,54 @@ func (dn *Daemon) checkStateOnFirstRun() error {
 	return dn.triggerUpdateWithMachineConfig(state.currentConfig, state.desiredConfig)
 }
 
+func forceFileExists() bool {
+	_, err := os.Stat(constants.MachineConfigDaemonForceFile)
+
+	// No error means we could stat the file; it exists
+	if err == nil {
+		return true
+	}
+
+	// We know the file doesn't exist
+	if goErrors.Is(err, os.ErrNotExist) {
+		return false
+	}
+
+	// We couldn't stat the file for other reasons, assume it doesn't exist
+	return false
+}
+
+func (dn *Daemon) maybeValidateOnDiskState(expectedConfig *mcfgv1.MachineConfig) (bool, error) {
+	// Skip validation
+	if forceFileExists() {
+		glog.Infof("Skipping on-disk validation; %s present", constants.MachineConfigDaemonForceFile)
+		dn.recorder.Eventf(getNodeRef(dn.node), corev1.EventTypeNormal, "OnDiskStateValidationSkipped", "On-disk configuration state validation skipped; %s present.", constants.MachineConfigDaemonForceFile)
+		return true, nil
+	}
+
+	// Don't skip validation
+	err := dn.runValidateOnDiskState(expectedConfig)
+
+	return false, err
+}
+
+func (dn *Daemon) runValidateOnDiskState(expectedConfig *mcfgv1.MachineConfig) error {
```

Comment:
I see you have many layering of this function, namely:
```
validateCurrentOnDiskStateInSyncNode
validateCurrentOnDiskState
maybeValidateOnDiskState
runValidateOnDiskState
validateOnDiskState
```
Do you think it would be possible to simplify that down a bit? Maybe to merge some of the functions and do some ifs inside? It feels a bit confusing to follow so many different layers of validation.